distributedtracing/distributedtracing.go

```diff
@@ -0,0 +1,60 @@
+package distributedtracing
+
+import (
+	"context"
+
+	"github.com/open-policy-agent/opa/logging"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+)
+
+// Enabled defines whether OpenTelemtry distributed tracing is enabled
+var Enabled bool
+
+func InitTracer(ctx context.Context, address string, serviceName string, sampleRatePercentage int, logger logging.Logger) (*trace.TracerProvider, error) {
+	logger.Info("Initializing tracer")
+	traceExporter, err := otlptracegrpc.New(ctx,
+		otlptracegrpc.WithInsecure(),
```

Comment:
Is this what everyone does? I could imagine wanting to have a TLS option here 🤔

Reply:
Added following flags to support tls
```
--distributed-tracing-tls {off,tls,mtls}            set distributed tracing tls scheme (default off)
--distributed-tracing-tls-cert-file string          set path of TLS certificate file of the tracing exporter
--distributed-tracing-tls-private-key-file string   set path of TLS private key file of the tracing exporter
--distributed-tracing-tls-skip-verify               disable certificate chain verification
```